ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/load/table/LoadTable.java

```diff
@@ -206,7 +207,13 @@ private void newTableTasks(ImportTableDesc tblDesc, Task<?> tblRootTask) throws
 
   private String location(ImportTableDesc tblDesc, Database parentDb)
       throws MetaException, SemanticException {
-    if (!tableContext.waitOnPrecursor()) {
+
+
+    if (tblDesc.tableType().equals(TableType.EXTERNAL_TABLE)) {
+      // For an external table, we need to use the path specified in the source table but the
+      // scheme, authority and root path of the target file system.
+      return context.warehouse.getDnsPath(tblDesc.getTableLocationPath()).toString();
```

Comment:
For Hive 3, the external table uses the location inside default external table directory (hive.metastore.warehouse.external.dir) if SET LOCATION is not done while creating external tables.
In this case, the path would be <default_external_tables_root>/<db_name>.db/<table_name>. But when we replicate this table to target, then the db_name in the path should use the target DB name as we allow table name during REPL LOAD. Also, there is no need to preserve the source path as it was set by Hive itself.

Reply:
When external tables on multiple sources have same location path and are replicated to a single target, they would overwrite each other's data. In order to avoid that we have discussed about having a configuration of base directory per source-target pair. All the external tables from a given source will be located under the base directory specified for that source. The location of the external table under this base directory will be same as that on the source. In such a scheme we can not honour default external table location. Hence for now, we are keeping the location on source and target same.